### PR TITLE
[WFLY-19577] Remove usage of AbstractAddStepHandler.attributes from u…

### PR DESCRIPTION
**Pull request by yersan**
…ndertow ServerAdd

Jira issue: https://issues.redhat.com/browse/WFLY-19577

### CODE REVIEW

File: undertow/src/main/java/org/wildfly/extension/undertow/ServerAdd.java [hunk 1]
```diff
@@ -92,7 +94,11 @@ protected void recordCapabilitiesAndRequirements(OperationContext context, Model
         // We currently don't have any attributes that reference capabilities, but we have this code in case that changes
         // since we are not calling the superclass code.
         ModelNode model = resource.getModel();
-        for (AttributeDefinition ad : getAttributes()) {
+
+        Map<String, AttributeAccess> attributeAccessMap = context.getResourceRegistration().getAttributes(PathAddress.EMPTY_ADDRESS);
+        for (Map.Entry<String, AttributeAccess> entry : attributeAccessMap.entrySet()) {
+            AttributeAccess attributeAccess = entry.getValue();
+            AttributeDefinition ad = attributeAccess.getAttributeDefinition();
```
Comment (pferraro) [inline undertow/src/main/java/org/wildfly/extension/undertow/ServerAdd.java:101]: Technically, we should skip requirement recording for runtime attributes and aliases.
See: https://github.com/wildfly/wildfly-core/blob/main/controller/src/main/java/org/jboss/as/controller/AbstractAddStepHandler.java#L259
I don't see any reason not to replace the first half of this method with a call to the super implementation, do you?

**Review by pferraro — CHANGES_REQUESTED** [review]

File: undertow/src/main/java/org/wildfly/extension/undertow/ServerAdd.java (hunk 1 above)
Comment (yersan) [reply undertow/src/main/java/org/wildfly/extension/undertow/ServerAdd.java:101]: @pferraro It looks like this method Java Doc responds to your question:

https://github.com/wildfly/wildfly/pull/18076/files#diff-9274d3139c44afa3e06390c846378a7a27d2b136ed92ac1050b193b86d0063e4R81-R83

Comment (pferraro) [reply undertow/src/main/java/org/wildfly/extension/undertow/ServerAdd.java:101]: ok.

**Review by pferraro — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @yersan and @pferraro

**Comment by bstansberry** [comment]
@yersan Can you send up a 33.x PR for this? It looks like this fixes a regression in WF 33, as that's when the call to the deprecated constructor was removed.

**Comment by yersan** [comment]
> Can you send up a 33.x PR for this?

Done, follow the related link above

**Pull request merged**